joblib/test/test_memory.py

```diff
@@ -162,14 +162,12 @@ def name_collision(x):
 
     b = name_collision
 
-    with warnings.catch_warnings(record=True) as w:
-        # Cause all warnings to always be triggered.
-        warnings.simplefilter("always")
+    with warns(JobLibCollisionWarning) as warninfo:
```

Comment:
I haven't neither seen elsewhere, nor have myself followed any strict convention with this name particularly. So I guess pytest's standard name `record` is just fine to use.